packages/backend-core/src/db/utils.ts

```diff
@@ -126,7 +128,11 @@ export async function getAppsByIDs(appIds: string[]) {
   )
   // have to list the apps which exist, some may have been deleted
   return settled
-    .filter(promise => promise.status === "fulfilled")
+    .filter(
+      promise =>
+        promise.status === "fulfilled" &&
+        (promise.value as DeletedApp).state !== AppState.INVALID
```

Comment:
Probably worth exposing a type helper for checking `state`. 
i.e. 

```ts
export function isDeletedApp = (
  app: App | DeletedApp
): App is DeletedApp => {
  if (app.state === AppState.INVALID) {
    return true
  }
  return false
}
```
usage:
```ts
promise.status === "fulfilled" && !isDeletedApp(promise.value)
```
This can then be reused across both checks

Reply:
`isDeletedApp` / `DeletedApp` / `AppState` should probably all be moved to:
`budibase/packages/types/src/sdk/apps.ts`

Reply:
Hmm - given this is essentially used in the context of only one function (albeit in 2 places for `pro`) it's a bit of a needless abstraction IMO using the type helper - the cast is valid (and clear) here given the type check here is only for one field and TS is being overly cautious. I'd still need to add the explicit field check in the type helper for TS to compile, which seems a bit overkill given it's usage:

```js
export function isDeletedApp(app: App | DeletedApp): app is DeletedApp {
  if ('state' in app && app.state === AppState.INVALID) {
    return true;
  }
  return false;
}
```